VIPER-SWIFT/Classes/Common/Categories/NSCalendar+CalendarAdditions.swift

```diff
@@ -8,129 +8,130 @@
-        let result = dateComponents.weekOfYear == duringSameWeekComponents.weekOfYear
+    func isDate(_ date: Date, duringSameWeekAsDate: Date) -> Bool {
+        let dateC = dateComponents([.weekOfYear], from: date)
+        let duringSameWeekComponents = dateComponents([.weekOfYear], from: duringSameWeekAsDate)
+        let result = dateC.weekOfYear == duringSameWeekComponents.weekOfYear
         return result
     }
     
-    func isDate(date: NSDate, duringWeekAfterDate: NSDate) -> Bool {
+    func isDate(_ date: Date, duringWeekAfterDate: Date) -> Bool {
         let nextWeek = dateForEndOfFollowingWeekWithDate(duringWeekAfterDate)
-        let dateComponents = components(.WeekOfYear, fromDate: date)
-        let nextWeekComponents = components(.WeekOfYear, fromDate: nextWeek)
-        let result = dateComponents.weekOfYear == nextWeekComponents.weekOfYear
+        let dateC = dateComponents([.weekOfYear], from: date)
+        let nextWeekComponents = dateComponents([.weekOfYear], from: nextWeek)
+        let result = dateC.weekOfYear == nextWeekComponents.weekOfYear
         return result
     }
     
-    func compareYearMonthDay(date: NSDate, toYearMonthDay: NSDate) -> NSComparisonResult {
+    func compareYearMonthDay(_ date: Date, toYearMonthDay: Date) -> ComparisonResult {
         let dateComponents = yearMonthDayComponentsFromDate(date)
         let yearMonthDayComponents = yearMonthDayComponentsFromDate(toYearMonthDay)
         
-        var result = compareInteger(dateComponents.year, right: yearMonthDayComponents.year)
+        var result = compareInteger(dateComponents.year!, right: yearMonthDayComponents.year!)
         
-        if result == NSComparisonResult.OrderedSame {
-            result = compareInteger(dateComponents.month, right: yearMonthDayComponents.month)
+        if result == ComparisonResult.orderedSame {
+            result = compareInteger(dateComponents.month!, right: yearMonthDayComponents.month!)
             
-            if result == NSComparisonResult.OrderedSame {
-                result = compareInteger(dateComponents.day, right: yearMonthDayComponents.day)
+            if result == ComparisonResult.orderedSame {
+                result = compareInteger(dateComponents.day!, right: yearMonthDayComponents.day!)
             }
         }
         
         return result
     }
     
-    func yearMonthDayComponentsFromDate(date: NSDate) -> NSDateComponents {
-        let newComponents = components(([.Year, .Month, .Day]), fromDate: date)
+    func yearMonthDayComponentsFromDate(_ date: Date) -> DateComponents {
+        let newComponents = dateComponents(([.year, .month, .day]), from: date)
         return newComponents
     }
     
-    func compareInteger(left: Int, right: Int) -> NSComparisonResult {
-        var result = NSComparisonResult.OrderedDescending
+    func compareInteger(_ left: Int, right: Int) -> ComparisonResult {
+        var result = ComparisonResult.orderedDescending
```

Comment:
It is possible. But we have to tell the compilator that result is a `ComparisonResult` variable:

> var result: ComparisonResult = .orderedDescending